lucene/analysis/common/src/java/org/apache/lucene/analysis/hunspell/CheckCompoundPattern.java

```diff
@@ -94,7 +94,7 @@ private boolean hasAllFlags(Root<?> root, char[] flags) {
   CharsRef expandReplacement(CharsRef word, int breakPos) {
     if (replacement != null && charsMatch(word, breakPos, replacement)) {
       return new CharsRef(
-          word.subSequence(0, breakPos)
+          new String(word.chars, 0, word.offset + breakPos)
```

Comment:
Include all compounds parts from the very beginning for the case check to work later